src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -101,13 +101,17 @@ public Checksum CreateChecksum(AnalyzerReference reference, CancellationToken ca
                         WriteUnresolvedAnalyzerReferenceTo(unresolved, writer);
                         break;
 
+                    case AnalyzerReference analyzerReference when analyzerReference.GetType().FullName == "Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioUnresolvedAnalyzerReference":
+                        WriteUnresolvedAnalyzerReferenceTo(analyzerReference, writer);
+                        break;
```

Comment:
either we ignore or throw or serialize, we still need case to determine specific reference type for now. if we need the case anyway, I think ignore or serialize won't matter that much as long as it is not throwing.